rskj-core/src/main/java/co/rsk/config/ConfigLoader.java

```diff
@@ -109,8 +110,16 @@ private Config getNetworkDefaultConfig(Config userConfig) {
                 return ConfigFactory.load(TESTNET_RESOURCE_PATH);
             } else if (NodeCliFlags.NETWORK_REGTEST.getName().equals(network)) {
                 return ConfigFactory.load(REGTEST_RESOURCE_PATH);
+            } else if (NodeCliFlags.NETWORK_DEVNET.getName().equals(network)) {
+                return ConfigFactory.load(DEVNET_RESOURCE_PATH);
             } else {
-                logger.warn("Invalid network '{}', using mainnet by default", network);
+                String exceptionMessage = String.format(
+                        "%s is not a valid network name (%s property)",
+                        network,
+                        SystemProperties.PROPERTY_BC_CONFIG_NAME
+                );
+                logger.warn(exceptionMessage);
+                throw new IllegalArgumentException(exceptionMessage);
```

Comment:
If the `blockchain.config.name` value is **mainnet** it'll fail, please modify this